src/Mvc/Mvc.RazorPages/src/ApplicationModels/PageConventionCollection.cs

```diff
@@ -46,7 +46,7 @@ internal MvcOptions MvcOptions
             get
             {
                 // Avoid eagerly getting to the MvcOptions from the options setup for RazorPagesOptions.
-                _mvcOptions ??= _serviceProvider.GetRequiredService<IOptions<MvcOptions>>().Value;
+                _mvcOptions ??= _serviceProvider!.GetRequiredService<IOptions<MvcOptions>>().Value;
```

Comment:
bit sketchy since one of the ctors sets this to null.

Reply:
Yup. I think this only gets called if the other ctor is used but 🙏🏽